superset-frontend/src/features/datasets/AddDataset/LeftPanel/index.tsx

```diff
@@ -122,6 +123,9 @@ export default function LeftPanel({
   datasetNames,
 }: LeftPanelProps) {
   const { addDangerToast } = useToasts();
+  const location = useLocation();
+
+  const isAddingDataSet = location.pathname.includes('/dataset/add') ?? false;
```

Comment:
This is not the correct way of handling this case. This component receives the dataset as a parameter and it should be the responsibility of the caller to update the dataset accordingly, which will trigger a re-render of this component. If you check the call stack, you'll see the dataset is coming from a hook. We need to invalidate the cached results of that hook if the database changes. @justinpark

Reply:
i see, i will check if i can patch a fix on the backend